.clang-format

```diff
@@ -0,0 +1,22 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Could you add some instructions into the README file about how to execute clang checks/rewrites?

Reply:
How about adding a pre-commit file?

Reply:
Can we integrate a Github action like https://github.com/cpp-linter/cpp-linter-action? FYI I have filed an INFRA JIRA to whitelist it but it seems not resolved yet: https://issues.apache.org/jira/browse/INFRA-26318

I have also added a C++ format check via Github action: https://github.com/apache/orc/blob/d7d6924de13fcbf938d953935b81949593969f12/.github/workflows/build_and_test.yml#L161-L176 But I'm in favor of `cpp-linter/cpp-linter-action` above.

cc @raulcd

Reply:
I agree we should have a pre-commit file and Github action. I use .clang-format for auto format when source file edited in vscode. TBH I don't know how to set a pre-commit and cpp-linter-action, so do you mind we did that in a seperate PR?

Reply:
I am not familiar with the `cpp-linter-action` or the `clang-format-action` but they don't seem complex to integrate. At Arrow we use pre-commit as suggested by @pitrou .

https://github.com/apache/arrow/blob/main/.pre-commit-config.yaml#L61-L101

https://github.com/apache/arrow/blob/main/.github/workflows/dev.yml#L43-L67

I am happy to integrate any, should be pretty straight forward once we agree on a configuration.

Reply:
pre-commit or cpp-linter-action will be integrated in later PR, so I will mark this as resolved, thanks.